src/main/java/Duke.java

```diff
@@ -1,10 +1,248 @@
+            checkSum(sum);
+            tasks[sum]=new Event(answer);
+            sum++;
+            numList(sum);
+        }
+        catch(ArrayIndexOutOfBoundsException e){
+            System.out.println("\n\t☹ OOPS!!! The description of an event cannot be empty.\n");
+        }
+        catch(exceedNum e){
+            System.out.println("\n\tThe list of tasks is full.\n");
+        }
+        return sum;
+    }
+
+    public static int processDeadline(String answer, Task[] tasks, int sum) {
+        try{
+            checkSum(sum);
+            tasks[sum]=new Deadline(answer);
+            sum++;
+            numList(sum);
+        }
+        catch(ArrayIndexOutOfBoundsException e){
+            System.out.println("\n\t☹ OOPS!!! The description of a deadline cannot be empty.\n");
+        }
+        catch(exceedNum e){
+            System.out.println("\n\tThe list of tasks is full.\n");
+        }
+        return sum;
+    }
+
+    public static int processTodo(String answer, Task[] tasks, int sum) {
+        try{
+            checkSum(sum);
+            tasks[sum]=new Todo(answer);
+            sum++;
+            numList(sum);
+        }
+        catch(ArrayIndexOutOfBoundsException e){
+            System.out.println("\n\t☹ OOPS!!! The description of a todo cannot be empty.\n");
+        }
+        catch(exceedNum e){
+            System.out.println("\n\tThe list of tasks is full.\n");
+        }
+        return sum;
+    }
+
+    public static void printList(Task[] tasks, int sum) {
+        System.out.println("\tHere are the tasks in your list:");
+        for (int i = 0; i< sum; i++) {
+            System.out.println("\t" + (i + 1) + "." + tasks[i]);
+        }
+        System.out.println();
+    }
+
+    public static void processDelete(Task[] tasks, String answer, int sum){
+        try {
+            String[] sentences = answer.split("delete ");
+            int index = Integer.parseInt(sentences[1]);
+            if (index>sum){
+                System.out.println("\n\tThe index exceed the range of tasks.\n");
+                return;
+            }
+            System.out.println("\n\tNoted. I've removed this task.");
+            System.out.println("\t  " + tasks[index - 1]);
+            for (int i = index - 1; i < sum - 1; i++) {
+                tasks[i] = tasks[i + 1];
+            }
+            tasks[sum - 1] = null;
+            System.out.println("\tNow you have " + (sum - 1) + " tasks in the list.\n");
+        }
+        catch (ArrayIndexOutOfBoundsException e){
+            System.out.println("\n\tPlease enter an index.\n");
+        }
+        catch (NumberFormatException e){
+            System.out.println("\n\tPlease enter a number as index.\n");
+        }
+    }
+
+    public static int readFile(Task[] tasks){
+        try {
+            int sum=0;
+            File f = new File("/Users/tissue/Desktop/CS2113T/ip/data/duke.txt");
+            Scanner s = new Scanner(f);
+            while(s.hasNext()){
```

Comment:
Aside from the inconsistent spacings, you can consider a `switch` block and then have each `case` block within mapped to a function. This way, you'll avoid long and verbose code in your `while` block.